src/odb/src/db/dbScanInst.cpp

```diff
@@ -116,5 +130,134 @@ dbOStream& operator<<(dbOStream& stream, const _dbScanInst& obj)
 //
 ////////////////////////////////////////////////////////////////////
 
+// User Code Begin dbScanInstPublicMethods
+void dbScanInst::setScanClock(std::string_view scan_clock)
+{
+  _dbScanInst* scan_inst = (_dbScanInst*) this;
+  scan_inst->scan_clock_ = scan_clock;
+}
+
+std::string_view dbScanInst::getScanClock() const
+{
+  _dbScanInst* scan_inst = (_dbScanInst*) this;
+  return scan_inst->scan_clock_;
+}
+
+void dbScanInst::setClockEdge(ClockEdge clock_edge)
+{
+  _dbScanInst* scan_inst = (_dbScanInst*) this;
+  scan_inst->clock_edge_ = static_cast<uint>(clock_edge);
+}
+
+dbScanInst::ClockEdge dbScanInst::getClockEdge() const
+{
+  _dbScanInst* scan_inst = (_dbScanInst*) this;
+  return static_cast<ClockEdge>(scan_inst->clock_edge_);
+}
+
+void dbScanInst::setBits(uint bits)
```

Comment:
What does 'bits' hold?  It is rather vague to me.

Reply:
The number of bits of the scan inst.

For simple flops it is just 1, for shift registers it could be the number of elements, for multibit cells with internal scan it would be the number of bits in the cell, etc.

Reply:
The seems rather overloaded.  At a minimum a descriptive comment is needed in db.h.